iOS Client/Store/MBStore.swift

```diff
@@ -61,13 +61,20 @@ class MBStore: NSObject {
                         print("There was an error downloading category data! \(unwrappedCatErr)")
                         completion(nil, catErr)
                     } else {
-                        self.downloadArticles(persistentContainer: persistentContainer, completion: { (isNewArticleData, articleErr) in
-                            isNewData = isNewAuthorData ?? false || isNewCategoryData ?? false || isNewArticleData ?? false
-                            if let unwrappedArticleErr = articleErr {
-                                print("There was an error downloading article data! \(unwrappedArticleErr)")
-                                completion(nil, articleErr)
+                        self.linkCategoriesTogether(persistentContainer: persistentContainer, completion: { (linkErr) in
+                            if let unwrappedLinkErr = linkErr {
+                                print("There was an error linking categories! \(unwrappedLinkErr)")
+                                completion(nil, linkErr)
```

Comment:
so now, during the data sync, after we get down all the categories, we have to go through them and set the parent relationship to point to the correct parent category (or nil if parentID is 0)